src/character/npc/NPC.js

```diff
@@ -0,0 +1,78 @@
+import RollFns from '../../RollFns';
+
+import Character from '../Character';
+
+export default class NPC extends Character {
+  constructor(config) {
+    super(config);
+
+    const {
+      actionRanks,
+      attributes,
+      baseHealth,
+      baseMana,
+      level,
+      name,
+      type,
+      weapon,
+    } = config;
+
+    this.actionRanks = actionRanks || {};
+    this.attributes = attributes;
+    this.level = level;
+    this.maxHealth = baseHealth + 2 * level + 5 * attributes.constitution;
+    this.maxMana = baseMana + attributes.willpower;
+    this.health = this.maxHealth;
+    this.mana = this.maxMana;
+    this.name = name;
+    this.type = type;
+    this.weapon = weapon;
+  }
+
+  canPerformAction(action) {
+    const actionName = action.name;
+    const manaCost = action.getManaCost(this.actionRanks[actionName]);
```

Comment:
Should this use `this.getActionRank(actionName)`? The difference is that the method returns 1 if the action is not in the hash.

Reply:
yup this sounds right